pkgs/by-name/au/auto-epp/package.nix

```
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchFromGitHub, python3 }:
+let
+  executable = "auto-epp";
+in
+stdenv.mkDerivation rec {
+  pname = "auto-epp";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "jothi-prasath";
+    repo = "auto-epp";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-7sI8K+7ZAdzBN/XOzYQQZ1f9t+fFo6fcXYzX6abNyQ8=";
+  };
+
+  buildInputs = [ python3 ];
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm555 ${executable} $out/bin/${executable}
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    mainProgram = executable;
```

I think extracting the name of the executable into it's own assignment is a bit unnecessary. It's not really something that should varies a lot, nor is it set conditionally.

```suggestion
stdenv.mkDerivation rec {
  pname = "auto-epp";
  version = "1.2.1";

  src = fetchFromGitHub {
    owner = "jothi-prasath";
    repo = "auto-epp";
    rev = "refs/tags/v${version}";
    hash = "sha256-7sI8K+7ZAdzBN/XOzYQQZ1f9t+fFo6fcXYzX6abNyQ8=";
  };

  buildInputs = [ python3 ];

  installPhase = ''
    runHook preInstall
    install -Dm555 auto-epp $out/bin/auto-epp
    runHook postInstall
  '';

  meta = with lib; {
    mainProgram = "auto-epp";
```